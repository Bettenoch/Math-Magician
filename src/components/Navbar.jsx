import React from 'react';
import '../styles/Navbar.css';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="nav-items">
    <div data-testid="mylogo" className="logo">Math Magicians</div>
    <ul className="nav-links">
      <li><Link className="navlink" to="/">Home</Link></li>
      <li><Link className="navlink" to="/Calculator">Calculator</Link></li>
      <li><Link to="/Quotes">Quotes</Link></li>
    </ul>
  </nav>
);

export default Navbar;
