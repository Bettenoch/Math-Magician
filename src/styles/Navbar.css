.nav-items {
  display: flex;
  justify-content: space-between;
  padding: 1.5rem 3rem;
  background-color: aliceblue;
}

.nav-links {
  display: flex;
  gap: 1.5rem;
}

.nav-links li {
  font-size: 1.2rem;
  font-weight: 600;
  padding-right: 0.5rem;
}

.navlink {
  border-right: 2px solid rgb(56, 9, 9);
  padding-right: 0.5rem;
}

.logo {
  font-size: 2rem;
  font-weight: 600;
}
