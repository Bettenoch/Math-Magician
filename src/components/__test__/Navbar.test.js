import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar snapshots', () => {
  it('should match the Navbar DOM snapshots', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test Home content', () => {
  it('should match the text content', () => {
    render(<Router><Navbar /></Router>);
    const mylogo = screen.getByTestId('mylogo');
    expect(mylogo).toHaveTextContent('Math Magicians');
  });
});
