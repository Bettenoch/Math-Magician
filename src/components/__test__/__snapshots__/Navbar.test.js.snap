// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshots should match the Navbar DOM snapshots 1`] = `
<nav
  className="nav-items"
>
  <div
    className="logo"
    data-testid="mylogo"
  >
    Math Magicians
  </div>
  <ul
    className="nav-links"
  >
    <li>
      <a
        className="navlink"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="navlink"
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/Quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
